src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs

```diff
@@ -518,10 +518,13 @@ private void WriteTo(AnalyzerReference reference, ObjectWriter writer, bool chec
                 writer.WriteString(nameof(AnalyzerFileReference));
                 writer.WriteInt32((int)SerializationKinds.FilePath);
 
-                if (!checksum)
+                if (checksum)
                 {
                     // we don't write full path when creating checksum
-                    writer.WriteString(file.FullPath);
+                    // make sure we always normalize assemblyPath to lower case since it comes from Assembly.Location
+                    // unlike FullPath which comes from string
+                    writer.WriteString(assemblyPath.ToLowerInvariant());
```

Comment:
We definitely tend to:

1. Preserve case for paths.
2. Use OrdinalIgnoreCase for comparisons with *actual* files from the disk/user inputs,
3. Use Ordinal for comparisons within Roslyn, in keeping with "1".